File: cpp/evaluate_reverse_polish_notation.cpp
class Solution {
public:
    int evalRPN(vector<string> &tokens) {
        stack<int> num_stack;
        for (int i=0; i<tokens.size(); ++i)
        {
            if (tokens[i]=="+" || tokens[i]=="-" || tokens[i]=="*" || tokens[i]=="/")
            {
                int num1 = num_stack.top();
                num_stack.pop();
                int num2 = num_stack.top();
                num_stack.pop();
                if (tokens[i]=="+") num2 += num1;
                else if (tokens[i]=="-") num2 -= num1;
                else if (tokens[i]=="*") num2 *= num1;
                else if (tokens[i]=="/") num2 /= num1;
                num_stack.push(num2);
            }
            else
                num_stack.push(stoi(tokens[i]));
        }
        int result = num_stack.top();
        return result;
    }
};

File: java/evaluate_reverse_polish_notation.java
import java.io.IOException;  
import java.util.Stack; 

public class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<Integer>();
        for (String token:tokens)
        {
            if (token.equals("+") || token.equals("-") || token.equals("*") || token.equals("/"))
            {
                int num2 = stack.pop();
                int num1 = stack.pop();
                if (token.equals("+")) num1 += num2;
                else if (token.equals("-")) num1 -= num2;
                else if (token.equals("*")) num1 *= num2;
                else if (token.equals("/")) num1 /= num2;
                stack.push(num1);
            }
            else
                stack.push(Integer.valueOf(token));
        }
        return stack.empty()?0:stack.pop();
    }
}